man/org.freedesktop.sysupdate1.xml

```diff
@@ -0,0 +1,442 @@
+      where it was enumerated. Possible values include: <literal>machine</literal> for containers and
+      virtual machines managed by
+      <citerefentry><refentrytitle>systemd-machined.service</refentrytitle><manvolnum>8</manvolnum></citerefentry>,
+      <literal>portable</literal> for <ulink url="https://systemd.io/PORTABLE_SERVICES">portable services</ulink>,
+      <literal>extension</literal> for system extensions managed by
+      <citerefentry><refentrytitle>systemd-sysext</refentrytitle><manvolnum>8</manvolnum></citerefentry>,
+      <literal>confext</literal> for configuration extensions managed by
+      <citerefentry><refentrytitle>systemd-confext</refentrytitle><manvolnum>8</manvolnum></citerefentry>,
+      <literal>component</literal> for components accepted by the <option>--component=</option> option of
+      <citerefentry><refentrytitle>systemd-sysupdate</refentrytitle><manvolnum>8</manvolnum></citerefentry>,
+      and <literal>host</literal> for the host system itself. At most one target will have a class of
+      <literal>host</literal>.</para>
+
+      <para>The <varname>Path</varname> property exposes more detail about where this target was found.
+      For <literal>machine</literal>, <literal>portable</literal>, <literal>extension</literal>, and
+      <literal>confext</literal> targets, this is the file path to the image. For <literal>component</literal>
+      and <literal>host</literal> targets, this is the name of a
+      <citerefentry><refentrytitle>sysupdate.d</refentrytitle><manvolnum>5</manvolnum></citerefentry>
+      directory.</para>
+
+      <para>The <varname>Name</varname> property exposes the name of this target. Note that the name is
+      unique within a class but is not necessarily unique between classes. For instance, it is possible
+      to have both a <literal>portable</literal> target named <literal>foobar</literal> and an
+      <literal>extension</literal> target named <literal>foobar</literal>, but it is not possible to have
+      two <literal>portable</literal> targets named <literal>foobar</literal>.</para>
+
+    </refsect2>
+
+    <refsect2>
+      <title>Security</title>
+
+      <para>Method calls on this service are authenticated via
+      <ulink url="https://www.freedesktop.org/software/polkit/docs/latest/">polkit</ulink>.</para>
+
+      <para><function>List()</function>, <function>Describe()</function>, and <function>CheckNew()</function>
+      use the polkit action <interfacename>org.freedesktop.sysupdate1.check</interfacename>.
+      By default, this action is permitted without administrator authentication.</para>
+
+      <para><function>Update()</function> uses the polkit action
+      <interfacename>org.freedesktop.sysupdate1.update</interfacename> when no version is specified.
+      By default, this action is permitted without administrator authentication. When a version is
```

Comment:
What's the rationale for allowing this to be called without auth by default? Given it changes the state of the system, isn't that risky? I understand you cannot tell it what to update to, but still, that is only a partial mitigation?

Reply:
Ok, seems there was some misunderstanding on my part of all the different components in play.

So here is some prior art:
* packagekit allows updating the system by default to [anyone with an active session](https://github.com/PackageKit/PackageKit/blob/e91426a53c30a807a8f94d48c245d40e941f49cd/policy/org.freedesktop.packagekit.policy.in#L173)
  * At least on Debian, this is [patched](https://sources.debian.org/patches/packagekit/1.3.0-1/policy.diff/#L39) so that it requires authentication by default, but has a rule so that an "administrative user that is currently at the computer" doesn't require authentication.
* eos-updater doesn't have polkit integration, and just uses a [dbus policy](https://github.com/endlessm/eos-updater/blob/master/eos-updater/com.endlessm.Updater.conf) to only allow user's that are at the console to talk to it.
* rpm-ostree [requires admin authentication](https://github.com/coreos/rpm-ostree/blob/main/src/daemon/org.projectatomic.rpmostree1.policy#L62) to update, but Fedora ships a [rules file](https://src.fedoraproject.org/rpms/fedora-release/blob/rawhide/f/org.projectatomic.rpmostree1.rules) to allow an "administrative user that is currently at the computer" to perform an update without entering their password.

Reply:
Ok, if there is prior art then I guess it's fine to do the same. It's worth noting it in a comment somewhere, maybe as a comment in the policy file itself

Reply:
My reasoning for this was that it's what distros will need to set it to anyways. Updating to the latest security patch must be encouraged as much as possible. 

All the situations I could think of where it doesn't make sense to have this allowed by default are corporate, and thus either using a package-based distro anyway, or are using some fleet update management software that probably isn't sysupdate. And even if using sysupdate, companies should be running their own update mirrors and publishing updates there only after vetting them.

So, realistically, the only situation where this setting is incorrect is for the one unicorn company that is using sysupdate instead of some fleet version management software, vets all OS updates, but doesn't run its own update mirrors. Thus, any time an update is approved for deployment an admin has to log in onto each system to apply it.

That company's admins can override the setting. In every other situation, allowing an update is either the correct solution, or it is completely irrelevant.